src/profanity.c

```diff
@@ -405,6 +409,9 @@ _shutdown(void)
     if (saved_status) {
         free(saved_status);
     }
+    //FIXME: kill all threads
```

Comment:
It is not very easy without skipping some memory free - so I would simply let the main thread keep the lock, let all threads running and exit (which will kill all threads and free all memory)